src/StackExchange.Redis.Extensions.Core/StackExchangeRedisCacheClient.cs

```diff
@@ -628,8 +628,13 @@ public IEnumerable<string> SearchKeys(string pattern)
 
 			foreach (var endpoint in endPoints)
 			{
-				var dbKeys = Database.Multiplexer.GetServer(endpoint).Keys(Database.Database, pattern);
+				var server = Database.Multiplexer.GetServer(endpoint);
+				if (!server.IsConnected || !server.Features.Scan)

```

Comment:
@EugeneKrapivin also core libraries has several `#if` ([here](https://github.com/aspnet/Mvc/blob/a78f77afde003c4a3fcf5dd7b6dc13dd9c85f825/src/Microsoft.AspNetCore.Mvc.Core/Internal/AmbiguousActionException.cs) and example).
If we want add dependencies without lot of `#if`we can create our logging factory but it looks like "reinvent the wheel", I don't like so much this way.

@aishelm your second solution is strictly related to the logging problem. Maybe we can add a parameter into the configuration to manage that scenario, something like `throwExceptionInCaseOfUnconnectedServer` and use this in all scenario who we need to check if the server is connected (just this right now).

Does it make sense for you?